packages/ui/src/modal/buy/BuyModalRenderer.tsx

```diff
@@ -328,76 +317,24 @@ export const BuyModalRenderer: FC<Props> = ({
     open,
     client,
     address,
-    tokenId,
-    contract,
-    is1155,
+    token,
     orderId,
     normalizeRoyalties,
     rendererChain,
-    paymentCurrency?.address,
-    paymentCurrency?.chainId,
+    rendererChain?.paymentTokens,
   ])
 
   useEffect(() => {
-    // ensure the tokens api has been fetched first
-    if (token) {
+    if (token || orderId) {
       fetchPath()
     }
-  }, [fetchPath, token])
+  }, [fetchPath, token, orderId])
 
-  const getCurrencyDetails = useCallback(() => {
-    if (orderId) {
-      return {
-        currency: listing?.price?.currency?.contract,
-        decimals: listing?.price?.currency?.decimals,
-        symbol: listing?.price?.currency?.symbol,
-        chainId: rendererChain?.id || 1,
-      }
-    } else if (is1155) {
-      return {
-        currency: path?.[0]?.currency,
-        decimals: path?.[0]?.currencyDecimals,
-        symbol: path?.[0]?.currencySymbol,
-        chainId: rendererChain?.id || 1,
-      }
-    } else {
-      return {
-        currency: token?.market?.floorAsk?.price?.currency?.contract,
-        decimals: token?.market?.floorAsk?.price?.currency?.decimals,
-        symbol: token?.market?.floorAsk?.price?.currency?.symbol,
-        chainId: rendererChain?.id || 1,
-      }
-    }
-  }, [orderId, is1155, listing, path, token, rendererChain])
-
-  // Set paymentCurrency to first paymentToken
   useEffect(() => {
-    if (paymentTokens[0] && listingCurrency && !paymentCurrency) {
+    if (paymentTokens[0] && !paymentCurrency) {
```

Comment:
I think we need to check that `fetchedInitialPath` is true, otherwise it will immediately set the `paymentCurrency` to the first `paymentToken` before we have the path

Reply:
Is it enough to just check the path here? to make sure it's available

Reply:
Yeah I think that works too